lib/date_time/date_time_modem.c

```diff
@@ -53,17 +54,18 @@ int date_time_modem_get(int64_t *date_time_ms)
 	 * "+CCLK: \"20/02/25,17:15:02+04\"\r\n\000{"
 	 */
 	rc = nrf_modem_at_scanf("AT+CCLK?",
-		"+CCLK: \"%u/%u/%u,%u:%u:%u",
+		"+CCLK: \"%u/%u/%u,%u:%u:%u%d",
 		&date_time.tm_year,
 		&date_time.tm_mon,
 		&date_time.tm_mday,
 		&date_time.tm_hour,
 		&date_time.tm_min,
-		&date_time.tm_sec
+		&date_time.tm_sec,
+		&tz
 	);
 
 	/* Want to match 6 args */
-	if (rc != 6) {
+	if (rc != 7) {
```

Comment:
Modem time can be set using `%CCLK` without specifying the time zone (value `+99` means that the time zone is "not set" or "unknown"). This results in `+CCLK?` returning a string without the time zone. I'm not sure if the network can give time without specifying the time zone, but I think it would be good to accept both 6 and 7 parameters here. If the time zone is missing, `tz` should be `0`.

Reply:
done